scripts/inf-manual-destroy.sh

```diff
@@ -87,14 +87,41 @@ random_text=${storage_account_name: -5}
 echo "Resource group: $rg_name"
 echo "Random text: $random_text"
 
-#delete ao_ai connection
-echo "Deleting open ai connection..."
-appid=$(az ad sp list --filter "displayname eq 'infoasst-web-$random_text'" --output json | jq -r '.[].appId')
-echo "Deleting role assignment for app id: $appid"
-role_assignment_id=$(az role assignment list --resource-group $oa_ai_rg_name --query "[?principalName=='$appid'].id" -o tsv)
-echo "Role assignment id: $role_assignment_id"
-az role assignment delete --ids $role_assignment_id
-echo "Role assignment deleted."
+echo "aoai name: " $oa_ai_rg_name
+
+# Delete shared ao_ai role assignment
+if [ -z "$oa_ai_rg_name" ]; then
+    echo "No shared Azure OpenAI role assignment found."
+else
+    echo "Deleting shared Azure OpanAI role assignment..."
+    appid=$(az ad sp list --filter "displayname eq 'infoasst-web-$random_text'" --output json | jq -r '.[].appId')
+    echo "Deleting role assignment for app id: $appid"
+    role_assignment_id=$(az role assignment list --resource-group $oa_ai_rg_name --query "[?principalName=='$appid'].id" -o tsv)
+    echo "Role assignment id: $role_assignment_id"
+    az role assignment delete --ids $role_assignment_id
+    echo "Role assignment deleted."
+fi
+
+
+# Delete models in the RG being deleted as a pre-step to the RG delete
```

Comment:
This should NOT delete any models in an RG that is outside of the one created by Info Assistant. We don't want to break other applications on a shared instance. The removal of role assignments above is correct and should be done regardless of shared AOAI or not, but deletion of models should ONLY be done on AOAI instances created by Info Assistant deployments.

Reply:
Correct. the logic is as follows...

It checks if we specified a shared aoai instance - if we did, then it doesn't attempt to delete any models
If we didn't then it proceeds with the logic. 
It then identifies models in the core rg we specified we are deleting, then iterates over these and deletes them